src/kernels/MIOpenBatchNormBwdSpatial.cl

```diff
@@ -320,7 +320,7 @@ MIOpenBatchNormBwdSpatial(const __global _FLOAT* __restrict x_in,
 #if(MIO_BN_REM4)
     if(lid < MIO_BN_REM4)
     {
-        unsigned int remkey = lid + MIO_BN_LESS4;
+        unsigned int remkey = (lid << 2) + MIO_BN_LESS4;
```

Comment:
This may require a few more subtle changes to avoid overflow within float4 read.